simulator/bypass/data_bypass.h

```diff
@@ -148,18 +144,30 @@ class DataBypass
         auto get_bypass_direction( const BypassCommand& bypass_command) const
         {
             const auto bypassing_stage = bypass_command.get_bypassing_stage();
-            return static_cast<std::size_t>( bypassing_stage);
+            return static_cast<uint8>( bypassing_stage);
+        }
+
+        // transforms bypassing data if needed in accordance with destination register
+        static auto get_bypassing_data( const MIPSInstr& instr)
```

Comment:
IMO, that method suits MIPSInstr class better. It takes nothing from DataBypass class.

Reply:
I think adapt_bypassed_data can be placed there as well.

Reply:
Sorry, my thought was not right.